### ISSUE TITLE
Latest build crashes

### ISSUE THREAD

**Issue by GijsTimmers** [jariz/Dumpert#43]
Running 349952a8e5653092e2845fe3d05d15d6d35e2f28.

[Log](http://paste.ubuntu.com/14486118/)

**Comment by CytoDev** [comment]
I'm on it

**Comment by jariz** [comment]
What version of android are you on again?

**Comment by GijsTimmers** [comment]
Kitkat, 4.4.2

**Comment by CytoDev** [comment]
debug or release variant?

**Comment by GijsTimmers** [comment]
I just `git pull`ed it. I'm on the master branch.

**Comment by CytoDev** [comment]
Yeah, but the gradle now has two build variants. I don't know what your configured default is. Some older gradle plugins have issues with the `android.support.v7.widget.LinearLayoutManager` class that is used by the NavigationView

**Comment by GijsTimmers** [comment]
Gradle 2.8, I believe. How to check?

**Comment by CytoDev** [comment]
In Android-Studio it automatically applies from the build.gradle file. I'm unsure how other IDEs handle it. Assuming you're using Android-Studio and using the correct build variant, I think I've located the issue. I'll test it around on 4.4.2  
  
Care to share your AVD config of the device?

**Comment by GijsTimmers** [comment]
Sure, how can I do that?

**Comment by CytoDev** [comment]
You need to clean your project so all resources are rebuilt. Seems the easiest fix. If that doesn't work, something went wrong with including the ripple file, and I'll get to the bottom of it.

**Comment by GijsTimmers** [comment]
Cleaning the project, then running doesn't work.

**Comment by CytoDev** [comment]
That's strange... Can you verify that you have `login_ripple.xml` inside your drawable folder and if `res/layout/navigation_login.xml` doesn't give any errors?

**Comment by GijsTimmers** [comment]
Sure:

```
$ readlink -f login_ripple.xml 
/home/user/AndroidStudioProjects/Dumpert/Dumpert/src/main/res/drawable/login_ripple.xml
$ cat login_ripple.xml 
<?xml version="1.0" encoding="utf-8"?>
<shape
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle">

    <solid android:color="?colorAccent" />

</shape>
```

What do you mean by:

> Can you verify that `res/layout/navigation_login.xml` doesn't give any errors?

**Comment by CytoDev** [comment]
You would want to open the layout in an IDE that checks the xml based on the android version you choose.  
Some screenshots because why not:
<img src="https://cloud.githubusercontent.com/assets/5646869/12293821/5c915a66-b9f6-11e5-9821-216ac8fc6ab3.png" width="33%" /><img src="https://cloud.githubusercontent.com/assets/5646869/12293822/5c92b5f0-b9f6-11e5-91da-3a41dd77ac40.png" width="33%" /><img src="https://cloud.githubusercontent.com/assets/5646869/12293823/5c9337fa-b9f6-11e5-993f-d9cba5b713aa.png" width="33%" />

**Comment by KeizerDev** [comment]
As I see you like using your cli? 
Run:
```
$ ./gradlew installDebug # in the root of your project
```
And post the output of that @GijsTimmers

**Comment by GijsTimmers** [comment]
Sure, I will do that ASAP. I don't have my laptop with me at the moment so
it'll take me an hour or so.

Op wo 13 jan. 2016 13:47 schreef Robert-Jan Keizer <notifications@github.com

**Comment by CytoDev** [comment]
@KeizerDev I'm writing that one down. Nifty!

**Comment by KeizerDev** [comment]
@CytoDev 
```
$ ./gradlew tasks 
```
:wink:

**Comment by CytoDev** [comment]
[The more you know...](https://media.giphy.com/media/wiQ9N7iMkMH2o/giphy.gif)

**Comment by GijsTimmers** [comment]
http://paste.ubuntu.com/14487185/

**Comment by KeizerDev** [comment]
So? You can close this issue. :wink:

**Comment by GijsTimmers** [comment]
Not really. I may have been unclear in the way I described the problem. The .apk installs fine. The problem is that I have a crash when opening Dumpert Reader.

**Comment by KeizerDev** [comment]
The app is installed on your phone you just have to start it and you have the most updated function. The gradle tasks `installDebug` only installs it, you still have to start it manually. I just wanted to know what the error was, but there doesn't seems to be one

**Comment by KeizerDev** [comment]
Ah, so your build tools are fine. We can conclude that. Can you show the output of `adb logcat` when firing the app.

**Comment by GijsTimmers** [comment]
http://paste.ubuntu.com/14487357/

**Comment by KeizerDev** [comment]
```
W/System.err(22181): 	... 51 more
W/dalvikvm(22181): threadid=1: calling UncaughtExceptionHandler
E/AndroidRuntime(22181): FATAL EXCEPTION: main
E/AndroidRuntime(22181): Process: io.jari.dumpert, PID: 22181
E/AndroidRuntime(22181): java.lang.RuntimeException: Unable to start activity ComponentInfo{io.jari.dumpert/io.jari.dumpert.activities.MainActivity}: android.view.InflateException: Binary XML file line #17: Error inflating class android.support.design.widget.NavigationView
E/AndroidRuntime(22181): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2338)
E/AndroidRuntime(22181): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2390)
E/AndroidRuntime(22181): 	at android.app.ActivityThread.access$800(ActivityThread.java:151)
E/AndroidRuntime(22181): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
E/AndroidRuntime(22181): 	at android.os.Handler.dispatchMessage(Handler.java:110)
E/AndroidRuntime(22181): 	at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime(22181): 	at android.app.ActivityThread.main(ActivityThread.java:5292)
E/AndroidRuntime(22181): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(22181): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(22181): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:824)
E/AndroidRuntime(22181): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:640)
E/AndroidRuntime(22181): 	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
E/AndroidRuntime(22181): 	at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime(22181): Caused by: android.view.InflateException: Binary XML file line #17: Error inflating class android.support.design.widget.NavigationView
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.createView(LayoutInflater.java:620)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:696)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:755)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
E/AndroidRuntime(22181): 	at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
E/AndroidRuntime(22181): 	at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:631)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(Native Method)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:397)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:353)
E/AndroidRuntime(22181): 	at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:256)
E/AndroidRuntime(22181): 	at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:109)
E/AndroidRuntime(22181): 	at io.jari.dumpert.activities.MainActivity.onCreate(MainActivity.java:53)
E/AndroidRuntime(22181): 	at android.app.Activity.performCreate(Activity.java:5264)
E/AndroidRuntime(22181): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1088)
E/AndroidRuntime(22181): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2302)
E/AndroidRuntime(22181): 	... 12 more
E/AndroidRuntime(22181): Caused by: java.lang.reflect.InvocationTargetException
E/AndroidRuntime(22181): 	at java.lang.reflect.Constructor.constructNative(Native Method)
E/AndroidRuntime(22181): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.createView(LayoutInflater.java:594)
E/AndroidRuntime(22181): 	... 26 more
E/AndroidRuntime(22181): Caused by: android.view.InflateException: Binary XML file line #40: Error inflating class Button
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:713)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:755)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
E/AndroidRuntime(22181): 	at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
E/AndroidRuntime(22181): 	at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:631)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(Native Method)
E/AndroidRuntime(22181): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:397)
E/AndroidRuntime(22181): 	at android.support.design.internal.NavigationMenuPresenter.inflateHeaderView(NavigationMenuPresenter.java:190)
E/AndroidRuntime(22181): 	at android.support.design.widget.NavigationView.inflateHeaderView(NavigationView.java:250)
E/AndroidRuntime(22181): 	at android.support.design.widget.NavigationView.<init>(NavigationView.java:172)
E/AndroidRuntime(22181): 	at android.support.design.widget.NavigationView.<init>(NavigationView.java:94)
E/AndroidRuntime(22181): 	... 29 more
E/AndroidRuntime(22181): Caused by: android.content.res.Resources$NotFoundException: File res/drawable/login_ripple.xml from drawable resource ID #0x7f020064
E/AndroidRuntime(22181): 	at android.content.res.Resources.loadDrawable(Resources.java:2152)
E/AndroidRuntime(22181): 	at android.content.res.TypedArray.getDrawable(TypedArray.java:602)
E/AndroidRuntime(22181): 	at android.content.res.XResources$XTypedArray.getDrawable(XResources.java:965)
E/AndroidRuntime(22181): 	at android.view.View.<init>(View.java:3579)
E/AndroidRuntime(22181): 	at android.widget.TextView.<init>
V/Provider/Settings(  695):  from settings cache , name = dropbox:data_app_crash , value = null
D/dalvikvm(  695): create interp thread : stack size=128KB
D/dalvikvm(  695): create new thread
```

**Comment by KeizerDev** [comment]
@CytoDev this is basically his error :wink:

**Comment by CytoDev** [comment]
Same error a before. Login ripple not found.

**Comment by KeizerDev** [comment]
`git reset --hard`

**Comment by CytoDev** [comment]
You are not installing the full package is the only reason I can think of. Maybe reset the emulator completely? Remove and reinstall the app? Clear the dalvik cache (if it's a real phone)?

**Comment by GijsTimmers** [comment]
> `git reset --hard`

Tried that, didn't work. Same problem.

**Comment by GijsTimmers** [comment]
I now stopped the app, removed its data. Then I removed the app. Rebooted into custom recovery, cleared Dalvik cache. Rebooted. Reinstalled via `./gradlew installDebug`. Problem still happening.

**Comment by CytoDev** [comment]
I'm taking my last guess by blaming `?colorAccent`. It should work even on older versions since it's defined in the theme, but then again knowing Android it might not be pointing at it correctly because the API is lower. Try changing the line containing `?colorAccent` in login_ripple.xml to see if my guess is right.

**Comment by GijsTimmers** [comment]
Change it in what? I have little experience with Java. Just comment it out?

**Comment by CytoDev** [comment]
I'll make a branch. Hold on tight.

**Comment by CytoDev** [comment]
@GijsTimmers You can go ahead and test [this branch](https://github.com/jariz/Dumpert/tree/testBranch43).

**Comment by GijsTimmers** [comment]
That fixed it.

**Comment by CytoDev** [comment]
Cool. I'll go ahead and update the ripple accordingly. Mind to test the next commit on that branch?

**Comment by GijsTimmers** [comment]
Sure, go ahead. Also, there's this white block thing going on. Not sure if it's supposed to be like that.
![screenshot_2016-01-13-16-33-15](https://cloud.githubusercontent.com/assets/972314/12298649/7e60a904-ba13-11e5-90b8-8b19f8cbc5d1.png)

**Comment by CytoDev** [comment]
Yes, that's because I changed the color of the element to white. It'll update soon enough.

**Comment by CytoDev** [comment]
@GijsTimmers can you test the last commit?

**Comment by GijsTimmers** [comment]
Yes -- it works now. The white block is also gone, it's just <kbd>Inloggen</kbd> now. Looking good!

**Comment by CytoDev** [comment]
Cool!

**Comment by CytoDev** [comment]
Resolved by #47.
